alibi_detect/saving/schemas.py

```diff
@@ -786,24 +757,26 @@ class SpotTheDiffDriftConfig(DriftDetectorConfig):
     Except for the `name` and `meta` fields, the fields match the detector's args and kwargs. Refer to the
     :class:`~alibi_detect.cd.SpotTheDiffDrift` documentation for a description of each field.
     """
+    p_val: float = .05
     binarize_preds: bool = False
     train_size: Optional[float] = .75
     n_folds: Optional[int] = None
     retrain_from_scratch: bool = True
     seed: int = 0
-    optimizer: Optional[Union[str, dict]] = None  # dict as can pass dict to tf.keras.optimizers.deserialize
+    optimizer: Optional[Union[str, OptimizerConfig]] = None
     learning_rate: float = 1e-3
     batch_size: int = 32
     preprocess_batch_fn: Optional[str] = None
     epochs: int = 3
     verbose: int = 0
     train_kwargs: Optional[dict] = None
-    dataset: str = '@alibi_detect.utils.tensorflow.data.TFDataset'
+    dataset: str = None
```

Comment:
I'm guessing the `TFDataset` wasn't the right thing to put here?

Reply:
It technically wasn't wrong because the config used to be associated with `SpotTheDiffTF`, whose kwarg was `dataset: Callable = TFDataset`. Now we (imo correctly) associate the config with `SpotTheDiff` class, whose kwarg is `dataset: Optional[Callable] = None`, hence the update. 

This is the same for all detectors with backends; the pydantic model fields now correspond to the top-level backend-agnostic class.

P.s. `Optional` was missing from the above. Have added!